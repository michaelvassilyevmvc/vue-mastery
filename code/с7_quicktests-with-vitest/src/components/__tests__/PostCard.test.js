// eslint-disable no-undef
import axios from 'axios'
import { mount, flushPromises } from '@vue/test-utils'
import PostCard from '@/components/PostCard.vue'

const mockPost = {
  userId: 1,
  id: 1,
  title: 'Post title 1',
  body: 'Simple post body...'
}

describe('Post Card Component', () => {
  test('created posts render correctly', () => {
    const title = 'Test Post'
    const body = 'test post body...'
    const wrapper = mount(PostCard, {
      props: { title, body }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
