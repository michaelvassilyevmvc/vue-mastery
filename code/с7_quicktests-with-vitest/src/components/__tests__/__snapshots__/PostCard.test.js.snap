// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Post Card Component > created posts render correctly 1`] = `
"<div title=\\"Test Post\\" body=\\"test post body...\\">
  <!--v-if-->
  <p data-testid=\\"loader\\">Loading...</p>
  <!--v-if-->
</div>"
`;
